### ISSUE TITLE
Check for age of IERS table fails with untrapped PolPMFlag lookup

### ISSUE THREAD

**Issue by bjweiner** [astropy/astroplan#436]
I'm having a lot of trouble getting astroplan to run after changing the location of the IERS table download to ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals2000A.all . An example error message is shown here:
`Traceback (most recent call last):
  File "mmtscheduler.py", line 210, in <module>
    import astroplan

  File "/usr/local/lib/python3.6/dist-packages/astroplan/__init__.py", line 32, in <module>
    get_IERS_A_or_workaround()

  File "/usr/local/lib/python3.6/dist-packages/astroplan/utils.py", line 57, in get_IERS_A_or_workaround
    iers.IERS.iers_table = _get_IERS_A_table()

  File "/usr/local/lib/python3.6/dist-packages/astroplan/utils.py", line 93, in _get_IERS_A_table
    index_of_last_observation = ''.join(table['PolPMFlag_A']).index('IP')

  File "/usr/local/lib/python3.6/dist-packages/astropy/table/table.py", line 1346, in __getitem__
    return self.columns[item]

  File "/usr/local/lib/python3.6/dist-packages/astropy/table/table.py", line 190, in __getitem__
    return OrderedDict.__getitem__(self, item)
KeyError: 'PolPMFlag_A'
`

Basically, in utils.py, _get_IERS_A_table() is trying to use the PolPMFlag_A key to figure out how old the table is, the table does not have this key, and an unrecoverable error is thrown. I am not sure if the missing key is a difference between the IERS table from ftp.cddis.nasa.gov and the previous file or something else. However, the relevant code (currently lines 94-98 of utils.py) should be wrapped in a try / except or something similar to catch the error.

**Comment by bjweiner** [comment]
I'm even getting this error when attempting to import astroplan. I don't need the IERS level of accuracy, but I need astroplan to run. I would really like to have some way of consistently running astroplan in low-accuracy mode without being dependent on these IERS files, because the current architecture of trying to download and verify them before falling back to low accuracy seems fragile. This may be an interdependency between astroplan and astropy.iers.

**Comment by bjweiner** [comment]
@tepickering and I just got this error to occur in a fresh Docker install with astropy 3.2.3 and astroplan 0.5. We could 'import astroplan' without an error, but when we run our scheduling code, it fails at the 'import astroplan' statement with the above error.

**Comment by tepickering** [comment]
i looked a little deeper and our code turned automatic downloading off and was not manually doing the download, either. removing the bits that turned that stuff off and letting astroplan/astropy do their default things under the hood got the IERS file to properly download and get properly parsed. 

note that we were still getting the error even after forcing a download from another python instance (e.g. by hand via interpreter), though. i think somehow bypassing the download also bypassed the finding/parsing of the IERS file which then generated the table KeyErrors. there's still some better checking to be done to handle this case more gracefully. i will try to cook up a test case to more cleanly reproduce what we saw and report back...

**Comment by bmorris3** [comment]
Hi @bjweiner, would you try the trick in https://github.com/astropy/astroplan/issues/427#issuecomment-547380768? Sometimes this error occurs when a corrupt or incomplete IERS file has been stored in the astropy cache.

**Comment by tepickering** [comment]
confirmed that i can reproduce this error reliably with this code:
```
#!/usr/bin/env python                                                                                                                  

from astropy.utils import iers
iers.IERS_A_URL = 'ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals.all'
iers.conf.auto_download = False

import astroplan
```
if there is a cached download of the IERS_A file. if the cache is cleared, then this will run without error. commenting out `iers.conf.auto_download = False` fixes the error as well. in our use case, however, we would like to disable `auto_download` so that we don't get tripped up by an unexpected download failure in the middle of an operational night like we did before...

**Comment by bjweiner** [comment]
I've been driving myself a little crazy trying to reproduce this because it is state-dependent. Thanks to Tim, I agree. Try this:
```python
from astropy.utils.data import clear_download_cache
clear_download_cache()
from astropy.utils import iers
iers.IERS_A_URL = 'ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals.all'
import astroplan
astroplan.download_IERS_A()
exit()

# Now we have a cached file.

python
from astropy.utils import iers
iers.IERS_A_URL = 'ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals.all'
iers.conf.auto_download = False
import astroplan
# PolPMFlag_A error occurs
exit()

# Disconnect network connection

from astropy.utils.data import clear_download_cache
clear_download_cache()
from astropy.utils import iers
iers.IERS_A_URL = 'ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals.all'
iers.conf.auto_download = False
import astroplan
astroplan.download_IERS_A()
# urllib.error.URLError occurs
exit()
```
I included the last, network connection error to show why it would be nice to fail more gracefully out of downloading the IERS file, although probably that should be attached to a separate issue.

**Comment by tepickering** [comment]
FTR, my testing was done using clean installs of the latest `astropy` and `astroplan` via `pip` within a docker container to assure a clean-slate environment...

**Comment by bmorris3** [comment]
I think one problem in the examples above by @tepickering and @bjweiner has to do with the URL that you're using. The IERS A table is at: `ftp://cddis.gsfc.nasa.gov/pub/products/iers/finals2000A.all` (see [here](https://github.com/astropy/astroplan/issues/425#issuecomment-547306406) for earlier discussion), it looks like you dropped the 2000A at the end.

**Comment by bjweiner** [comment]
Hi @bmorris3, you are absolutely right - the actual code that we are trying to run uses the finals2000A.all file, and somehow the 2000A got dropped in the examples. However, the same errors do occur when the finals2000A.all filename is used in the above examples. 

As I understand it, the finals.all and finals2000A.all files have essentially the same format, but use different models for the Earth precession/nutation. So using the wrong file might give wrong answers for the time offset, but it's not clear why it would cause this crash when the cached file exists (and not when it is downloaded fresh).

**Comment by bmorris3** [comment]
Ok I can confirm that following your instructions above I reproduce the missing `PolPMFlag_A` column error. I wonder if the problem is related to the fact that the file is old. Opening up the file from `ftp://cddis.gsfc.nasa.gov` reveals that the last date in the table is in 2017... Digging deeper.

**Comment by tepickering** [comment]
the issue is that if IERS_A is available, the returned table has the following columns:
```
In [3]: t.columns                                                                                                                      
Out[3]: <TableColumns names=('year','month','day','MJD','PolPMFlag_A','PM_x_A','e_PM_x_A','PM_y_A','e_PM_y_A','UT1Flag_A','UT1_UTC_A','e_UT1_UTC_A','LOD_A','e_LOD_A','NutFlag_A','dX_2000A_A','e_dX_2000A_A','dY_2000A_A','e_dY_2000A_A','PM_X_B','PM_Y_B','UT1_UTC_B','dX_2000A_B','dY_2000A_B','UT1_UTC','UT1Flag','PM_x','PM_y','PolPMFlag','dX_2000A','dY_2000A','NutFlag')>
```
if it is not available somehow, then the fallback is to use the bundled IERS_B table which has the following columns:
```
In [4]: t.columns                                                                                                                      
Out[4]: <TableColumns names=('year','month','day','MJD','PM_x','PM_y','UT1_UTC','LOD','dX_2000A','dY_2000A','e_PM_x','e_PM_y','e_UT1_UTC','e_LOD','e_dX_2000A','e_dY_2000A')>
```
so the logic needs to be fixed to handle both cases. in the latter case the index you want is simply `-1` since the bundled data is only historical up to Nov 2017 currently.
